lib/logidze/meta.rb

```diff
@@ -50,30 +79,46 @@ def call_block_in_meta_context
       ensure
         meta_stack.pop
       end
+    end
 
-      def current_meta
-        meta_stack.reduce(:merge) || {}
-      end
+    class MetaWithoutTransaction < MetaWrapper # :nodoc:
+      def perform
+        raise ArgumentError, "Block must be given" unless block
```

Comment:
Let's move this check to the base class. Don't know why we don't have it in the current implementation. @DmitryTsepelev ?

And we can make this `perform` the base implementation and only override `call_block_in_meta_context` in the transactional subclass, not `#perform`.

Reply:
Fair point, it does not make any sense to call `.with_meta` without a block